src/Type/NeverType.php

```diff
@@ -0,0 +1,81 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type;
+
+use PHPStan\TrinaryLogic;
+
+class NeverType implements CompoundType
+{
+
+	/**
+	 * @return string|null
+	 */
+	public function getClass()
+	{
+		return null;
+	}
+
+	/**
+	 * @return string[]
+	 */
+	public function getReferencedClasses(): array
+	{
+		return [];
+	}
+
+	public function combineWith(Type $otherType): Type
+	{
+		return $this;
+	}
+
+	public function accepts(Type $type): bool
+	{
+		return true;
+	}
+
+	public function describe(): string
+	{
+		return '*NEVER*';
```

Comment:
Food for thought - NeverType is almost the same as ErrorType right now. What about extending it from MixedType right now?

Reply:
NeverType is the absolute opposite of MixedType – NeverType being the [bottom type](https://en.wikipedia.org/wiki/Bottom_type) and MixedType being the [top type](https://en.wikipedia.org/wiki/Top_type). Therefore extending MixedType does not make any semantic sense.

Reply:
OK 👍